src/bigartm/cooccurrence_dictionary.cc

```diff
@@ -0,0 +1,442 @@
+// Copyright 2017, Additive Regularization of Topic Models.
+
+#include "cooccurrence_dictionary.h"
+
+#include <cstdio>
+#include <unordered_map>
+#include <string>
+#include <iostream>
+#include <cstdlib>
+#include <fstream>
+#include <map>
+#include <vector>
+#include <assert.h>
+#include <future>
+#include <mutex>
+#include <sstream>
+#include <iomanip>
+
+#include "boost/algorithm/string.hpp"
+#include "boost/filesystem.hpp"
+#include "boost/utility.hpp"
+
+using namespace std;
+using namespace boost;
+
+CooccurrenceBatch::CooccurrenceBatch(int batch_num, const char filemode, const std::string& disk_path) {
+  const int max_name_len = 30;
+  char num_str[max_name_len] = "";
+  sprintf(num_str,"%.3d", batch_num);
+  std::string filename = string(num_str);
+  filename += string(".bin");
+  boost::filesystem::path full_filename = boost::filesystem::path(disk_path)
+                 / boost::filesystem::path(filename);
+  if (filemode == 'w') {
+    out_batch.open(full_filename.string());
+    if (out_batch.bad()) {
+      std::cerr << "CooccurrenceBatch::CooccurrenceBatch: Failed to create batch\n";
+      throw 1;
+    }
+  } else {
+    in_batch.open(full_filename.string());
+    if (in_batch.bad()) {
+      std::cerr << "CooccurrenceBatch::CooccurrenceBatch: Failed to open existing batch\n";
+      throw 1;
```

Comment:
Everywhere you'd like to throw exception use BOOST_THROW_EXCEPTION macro and one of pre-defined artm exceptions